### PR TITLE
Test that changing <iframe srcdoc> processes attributes

### PR DESCRIPTION
**Pull request by shvaikalesh**
Blink bug: [srcdoc removal does not process iframe attributes](https://bugs.chromium.org/p/chromium/issues/detail?id=513689).
WebKit bug: [`<iframe>` attributes should be processed on "srcdoc" attribute removal](https://bugs.webkit.org/show_bug.cgi?id=205995).

### CODE REVIEW

File: html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  return iframe;
+}
+
+async_test(function(t) {
+  var iframe = createIFrameWithBlobSrc();
+  iframe.onload = t.step_func(function() {
+    assert_equals(iframe.contentDocument.location.protocol, "blob:");
+    assert_equals(iframe.contentDocument.body.textContent, "src");
+
+    iframe.onload = t.step_func(function() {
```
Comment (annevk) [inline html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html]: Nit: could use step_func_done

File: html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}, "Setting `srcdoc` triggers attributes processing");
+
+async_test(function(t) {
+  var iframe = createIFrameWithBlobSrc();
+  iframe.srcdoc = "old";
+  iframe.onload = t.step_func(function() {
+    assert_equals(iframe.contentDocument.location.href, "about:srcdoc");
+    assert_equals(iframe.contentDocument.body.textContent, "old");
+
+    iframe.onload = t.step_func(function() {
```
Comment (annevk) [inline html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html]: Nit: same

File: html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      t.done();
+    });
+
+    var isChanging = true;
+    iframe.srcdoc = "new";
+    isChanging = false;
+  });
+
+  document.body.appendChild(iframe);
+}, "Changing `srcdoc` triggers attributes processing");
```
Comment (annevk) [inline html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html]: Maybe "Setting srcdoc property ..."?

File: html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+async_test(function(t) {
+  var iframe = createIFrameWithBlobSrc();
+  iframe.srcdoc = "srcdoc";
+  iframe.onload = t.step_func(function() {
+    assert_equals(iframe.contentDocument.location.href, "about:srcdoc");
+    assert_equals(iframe.contentDocument.body.textContent, "srcdoc");
+
+    iframe.onload = t.step_func(function() {
```
Comment (annevk) [inline html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html]: Nit: same

File: html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    assert_equals(iframe.contentDocument.body.textContent, "src");
+
+    iframe.onload = t.step_func(function() {
+      assert_false(isSetting);
+      assert_equals(iframe.contentDocument.location.href, "about:srcdoc");
+      assert_equals(iframe.contentDocument.body.textContent, "srcdoc");
+      t.done();
+    });
+
+    var isSetting = true;
```
Comment (annevk) [inline html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html]: I guess these variable declarations are hoisted? Bit confusing to read, but probably fine.

**Review by annevk — APPROVED** [review]
Thanks for writing these, I'd like a slightly better name for the second test, but not going to block on that.

File: html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html (hunk 1 above)
Comment (shvaikalesh) [reply html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html]: ccc7251

File: html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html (hunk 2 above)
Comment (shvaikalesh) [reply html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html]: ccc7251

File: html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html (hunk 4 above)
Comment (shvaikalesh) [reply html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html]: ccc7251

File: html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html (hunk 5 above)
Comment (shvaikalesh) [reply html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html]: 8b8a334

File: html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html (hunk 3 above)
Comment (shvaikalesh) [reply html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html]: I've slightly improved test cases description in b764ed9, but kept "changing": attributes should not be processed after `iframe.srcdoc = iframe.srcdoc`, right?

Comment (annevk) [reply html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html]: Per the DOM Standard that is a change.

**Review by annevk — APPROVED** [review]
Thanks!

**Comment by jdm** [comment]
@annevk @shvaikalesh This test asserts that a srcdoc iframe's load event is fired before returning from the srcdoc attribute modification. Where in the specification is that requirement documented?

**Comment by annevk** [comment]
I missed that, that's indeed wrong.

@shvaikalesh would you be willing to change the relevant `assert_false`s to `assert_true`s?

File: html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html (hunk 3 above)
Comment (shvaikalesh) [reply html/semantics/embedded-content/the-iframe-element/srcdoc_process_attributes.html]: Thank you, resolved in https://github.com/web-platform-tests/wpt/pull/20084/commits/522f4c5bddb65c56b4440cf0cd0ed186edb37235.

**Pull request merged**